pkg/util/cache/cache.go

```diff
@@ -43,8 +44,8 @@ type cacheItem[K comparable, V any] struct {
 }
 
 type (
-	Loader[K comparable, V any]    func(key K) (V, bool)
-	Finalizer[K comparable, V any] func(key K, value V) error
+	Loader[K comparable, V any]    func(ctx context.Context, key K) (V, error)
+	Finalizer[K comparable, V any] func(ctx context.Context, key K, value V) error
 )
```

Comment:
two good points for adding context:
1. avoid pointless wait when context has been canceled
2. make log output able to be tracked by traceID